src/figuresClasses.ts

```diff
@@ -1,19 +1,104 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.color = color;
+
+    const sides = [a, b, c];
+    const longestSide: number = Math.max(...sides);
+    const indexOfLongest = sides.indexOf(longestSide);
+    const otherSides: number[] = sides.slice(0, indexOfLongest);
+    const arePositive: boolean = sides[0] > 0 && sides[1] > 0 && sides[2] > 0;
+    const isTriangle: boolean = longestSide < otherSides[0] + otherSides[1];
+
+    if (!arePositive || !isTriangle) {
+      throw new Error('Not a triangle');
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
 
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
```

Comment:
- Avoid one-letter names such as `p`.
- You do `this.a` and other twice, you can use destructing. `const { a, b, c } = this;`